pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/function/RangeAggregationFunction.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (C) 2014-2015 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.query.aggregation.function;
+
+import java.io.Serializable;
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.List;
+
+import java.util.Locale;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linkedin.pinot.common.Utils;
+import com.linkedin.pinot.common.data.FieldSpec.DataType;
+import com.linkedin.pinot.common.request.AggregationInfo;
+import com.linkedin.pinot.core.common.Block;
+import com.linkedin.pinot.core.common.BlockDocIdIterator;
+import com.linkedin.pinot.core.common.BlockSingleValIterator;
+import com.linkedin.pinot.core.common.Constants;
+import com.linkedin.pinot.core.query.aggregation.AggregationFunction;
+import com.linkedin.pinot.core.query.aggregation.CombineLevel;
+import com.linkedin.pinot.core.query.aggregation.function.RangeAggregationFunction.RangePair;
+import com.linkedin.pinot.core.query.utils.Pair;
+import com.linkedin.pinot.core.segment.index.readers.Dictionary;
+
+/**
+ * This function will take a column and do min/max and compute diff on that.
+ *
+ */
+public class RangeAggregationFunction implements AggregationFunction<RangePair, Double> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(RangeAggregationFunction.class);
+  private static final double DEFAULT_RANGE_VALUE = -1;
+
+  private String _rangeByColumn;
+
+  public RangeAggregationFunction() {
+
+  }
+
+  @Override
+  public void init(AggregationInfo aggregationInfo) {
+    _rangeByColumn = aggregationInfo.getAggregationParams().get("column");
+
+  }
+
+  @Override
+  public RangePair aggregate(Block docIdSetBlock, Block[] block) {
+    double min = Double.POSITIVE_INFINITY;
```

Comment:
In case of empty block, the range would be infinity. Do we want that or some other value (eg NaN)?